packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts

```diff
@@ -307,3 +308,31 @@ export const useUpdateKubeVersion = ({
     ...mutation,
   };
 };
+type TerminateClusterProps = {
+  projectId: string;
+  kubeId: string;
+  onError: (cause: Error) => void;
+  onSuccess: () => void;
+};
+
+export const useTerminateCluster = ({
+  projectId,
+  kubeId,
+  onError,
+  onSuccess,
+}: TerminateClusterProps) => {
+  const mutation = useMutation({
+    mutationFn: async () => terminateCluster(projectId, kubeId),
+    onError,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
```

Comment:
i think you need to invalidate the list of clusters too

Reply:
done